kernel/vmemitcodes.ml

```diff
@@ -78,13 +78,9 @@ let output buf n =
 let input s pos =
   let c = Char.code s.[!pos] in
   if Int.equal c 0x00 then
-    (* TODO: use String.get_int32_be when available in OCaml 4.13.0 *)
-    let c1 = Char.code s.[!pos + 1] in
-    let c2 = Char.code s.[!pos + 2] in
-    let c3 = Char.code s.[!pos + 3] in
-    let c4 = Char.code s.[!pos + 4] in
+    let c = String.get_int32_be s (!pos+1) in
     let () = pos := !pos + 5 in
-    (c1 lsl 24) lor (c2 lsl 16) lor (c3 lsl 8) lor c4
+    Int32.to_int c
```

Comment:
if the slowdown happens again I'll try without this change in case get_int32_be is somehow slow

Reply:
I just checked the assembly code. It looks much better with `get_int32_be`. In particular, it no longer performs four in-bound checks in a row. This should make the new code much faster, because, even if all the conditional branches in the old code are perfectly predicted, having so many branches in a row should still bring the processor to its knees by preventing speculation. That said, the new code is still quite ugly (and would never have been generated by compilers such as GCC or LLVM), so it is possibly triggering a processor defect:
```asm
	movslq	%eax, %rax
	bswap	%eax
	movslq	%eax, %rax
	movslq	%eax, %rax
```
(How many `movslq` is too many?)

Reply:
I doubt this is any relevant to the overall performance of Rocq, we're barely passing any time in this area of the code.